### PR TITLE
Ft login page bug fix

### PR DESCRIPTION
**Pull request by GilbertUp**
### What does this PR do?

It allows registered user to login

### How should this be tested?
-Clone this [repo](https://github.com/atlp-rwanda/Dominators-Phantom-FE/compare/develop...ft-login-page-bug-fix)
-checkout to the branch called ft-login-page-bud-fix
-Run the command npm i and later npm start
-click on login and in put admin7@test.com and pass12345 as user admin  credential
### What to expect?
-if the email is in invalid format or missing, an appropriate message will be displayed
-if the password is missing or less than 8 characters, an appropriate message will be displayed
-For successful attempt you will be redirected to admin dashboard and error messages are displayed for unsuccessful attempts
### Acceptance criteria

- [x] Appropriate error messages are displayed on invalid inputs (Form validation).
- [x] Appropriate error messages are displayed on unsuccessful attempts
- [x]  Upon successful login, the user should be redirected to their notification page (At this point in time, the notification page might be a black page).
- [ ]  Skeleton UIs/screen are/is shown when the page loads/is loading.
- [x]  The page is responsive
- [ ]  Each user  role upon successful login should be redirected to their respective profile page

### CODE REVIEW

**Review by aimelive — APPROVED** [review]

File: src/components/Login/LoginComponent.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+const LoginComponent = () => {
+  const [users,setUsers] = useState(null);
+  const [loading,setIsLoading] = useState(false);
+  const redirect = useNavigate();
+  useEffect(()=>{
+    
+  })
+
+  const handleLogin = (values)=>{
+    console.log(db)
```
Comment (furebo) [inline src/components/Login/LoginComponent.js]: Please remove the console.log line.

File: src/components/Login/LoginComponent.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+   fetch(`${db}/api/v1/users/login`, { 
+     method: "POST",
+     body: bodyContent,
+     headers: headersList
+   })
+    .then((res)=>{
+      return res.json();
+    })
+    .then ((result)=>{
+      console.log(result)
```
Comment (furebo) [inline src/components/Login/LoginComponent.js]: Please remove also the console.log line

**Review by furebo — CHANGES_REQUESTED** [review]
Remove the console.log lines and squash the commits to one commit

**Comment by Anguandia** [comment]
Please note the following:

@GilbertUp good PR description. Much as you are now in the backend, please address the following
- squash these commits, 38 is unacceptable
- was this a bug or a feature? the branch name is confusing. Please rename it following the team conventions and best practices documented [here](https://github.com/atlp-rwanda/engineering-playbook/tree/master/5.%20Developing/Conventions#branch-naming)
- despite the spot-on PR description, the formatting is still wanting, please acquaint yourself with markdown syntax. you can search online

@gerard1001  you now own this task, please respond to any technical review comments/change requests to this PR. Also address the following:
- write unit tests for the feature
- test setup is wanting, you may start from setting up testing

**Review by UmuhireJessie — APPROVED** [review]

**Comment by Anguandia** [comment]
A few more observations:
- please change the cursor to pointer when hovering over the login button
- it will be good to read the backend base url from the environment so that different instances may be used in different environments

File: src/components/Login/LoginComponent.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+          console.log("User", result.data.user.firstName);
+
+          if (
+            result.data.user.role == "admin" ||
+            result.data.user.role == "operator" ||
+            result.data.user.role == "driver"
+          ) {
+            redirect("/Dashboard");
+          } else {
+          }
```
Comment (furebo) [inline src/components/Login/LoginComponent.js]: Please remove the empty else block

**Review by furebo — CHANGES_REQUESTED** [review]
Hi good work Gelard but resolve the appearing conflicts do the requested change

File: src/utils/db.js [hunk 4]
```diff
@@ -1 +1 @@
-export const db = "http://localhost:4000";
+export const db = "http://localhost:4000/api/v1";
```
Comment (Anguandia) [inline src/utils/db.js]: We need to dynamically inject this from the environment

File: package.json [hunk 5]
```diff
@@ -33,8 +34,8 @@
     "redux-devtools-extension": "^2.13.9",
     "redux-state-sync": "^3.1.2",
     "redux-thunk": "^2.4.1",
-    "sweetalert": "^2.1.2",
-    "serve": "^13.0.2"
+    "serve": "^13.0.2",
+    "sweetalert": "^2.1.2"
   },
   "devDependencies": {
     "@babel/core": "^7.17.9",
```
Comment (Anguandia) [inline package.json:41]: We also need to streamline project vs dev dependencies

**Review by Anguandia — APPROVED** [review]
Good work! just the 2 changes, we'll be good to go

**Review by furebo — APPROVED** [review]

**Pull request merged**